app/Models/Chat/Channel.php

```diff
@@ -555,18 +557,26 @@ public function validationErrorsTranslationPrefix()
         return 'chat.channel';
     }
 
+    /**
+     * Unhides UserChannels as necessary when receiving messages.
+     *
+     * @return void
+     */
     private function unhide()
     {
-        if (!$this->isPM()) {
+        if (!$this->isHideable()) {
             return;
         }
 
-        return UserChannel::where([
-            'channel_id' => $this->channel_id,
-            'hidden' => true,
-        ])->update([
-            'hidden' => false,
-        ]);
+        $params = ['channel_id' => $this->channel_id, 'hidden' => true];
+
+        $users = User::whereIn('user_id', UserChannel::where($params)->select('user_id')->pluck('user_id'))->get();
```

Comment:
remove `pluck` so it's a subquery instead?

Reply:
Turns out I originally added the `pluck` because the `whereIn` didn't match anything with only subquery when running the tests 🤔